### PR TITLE
[PLAY-679] Selectable List to Typescript

### PR DESCRIPTION
**Pull request by nidaqg**
#### Screen

Screenshot 2023-04-03 at 2 22 30 PM

#### Breaking Changes

No

#### Runway Ticket URL

[PLAY-679](https://nitro.powerhrg.com/runway/backlog_items/PLAY-679)

#### How to test this

[INSERT TESTING DETAILS]

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_selectable_list/_selectable_list.tsx [hunk 1]
```diff
@@ -30,17 +30,18 @@ const SelectableList = (props: SelectableListProps) => {
   const classes = classnames(buildCss('pb_selectable_list_kit'), globalProps(props), className)
   const dataProps = buildDataProps(data)
   const isRadio = props.variant === "radio"
-  const defaultCheckedRadioValue = children.filter(item => item.props.defaultChecked)[0]?.props?.value
+  const defaultCheckedRadioValue = children.filter((item: {props:SelectableListItemProps} ) => item.props.defaultChecked)[0]?.props?.value
```
Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_selectable_list/_selectable_list.tsx:33]:
```suggestion
const defaultCheckedRadioValue = children.find(
  item => React.isValidElement(item) && item.props.defaultChecked
)?.props?.value
```

Comment (jasperfurniss) [reply playbook/app/pb_kits/playbook/pb_selectable_list/_selectable_list.tsx:33]: Can you see if that fixes your problem? 
I think by using `isValidElement` we can keep the children prop as React.ReactNode[]

File: playbook/app/pb_kits/playbook/pb_selectable_list/_selectable_list.tsx [hunk 2]
```diff
@@ -59,13 +61,13 @@ const SelectableList = (props: SelectableListProps) => {
         className={classes}
         id={id}
     >
-      <List {...props}>
+      <List variant={props.variant}>
```
[the remark this answers is not in the document]
Comment (jasperfurniss) [reply playbook/app/pb_kits/playbook/pb_selectable_list/_selectable_list.tsx:64]: Yeah makes sense. And we can ship this.

**Review by jasperfurniss — APPROVED** [review]

File: playbook/app/pb_kits/playbook/pb_selectable_list/_selectable_list.tsx (hunk 1 above)
Comment (nidaqg) [reply playbook/app/pb_kits/playbook/pb_selectable_list/_selectable_list.tsx:33]: Tested it out locally and it did not fix it :(

**Pull request merged**